src/models/game.ts

```diff
@@ -121,6 +132,26 @@ class Game extends Model<InferAttributes<Game>, InferCreationAttributes<Game>> {
       return playerNumberOrder[currentPlayerIndex + 1];
     }
   }
+
+  async toSnarkyGameState(): Promise<GameState> {
+    const pieces = await serializePiecesTreeFromGameId(this.id);
+    const arena = await serializeArenaTreeFromGameId(this.id);
+    const players = await this.gamePlayersInTurnOrder();
+    const p1 = await players[0].player();
+    const p2 = await players[1].player();
+    const turnsNonce = this.turnNumber || 0;
+    const currentPlayerTurn = (await this.turnGamePlayerNumber()) + 1;
```

Comment:
Almost all of this data comes from separate queries to various tables which is why I'm going to leave the work of extracting the logic for later.